crates/iota-transaction-builder/src/builder/mod.rs

```diff
@@ -702,6 +702,30 @@ impl<L> TransactionBuilder<Client, L> {
         let mut inputs = Vec::new();
         let mut gas = Vec::new();
         let mut input_map = HashMap::new();
+        if !self.data.inputs.values().any(|i| i.is_gas) {
+            for coin in self
+                .client
+                .objects(
+                    ObjectFilter {
+                        type_: Some(
+                            StructTag::coin(TypeTag::Struct(Box::new(StructTag::gas_coin())))
+                                .to_string(),
+                        ),
```

Comment:
```suggestion
                        type_: Some(StructTag::gas_coin().to_string()),
```
`StructTag::gas_coin()` already has everything, right now it doesn't work

Reply:
oops